### ISSUE TITLE
Fix `selector-pseudo-class-no-unknown` false positive for `:popover-open`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#7424]
### What minimal example or steps are needed to reproduce the bug?

```css
:popover-open {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
}
```

### What did you expect to happen?

no error

### What actually happened?

Unexpected unknown pseudo-class selector ":popover-open"

### Do you have a proposal to fix the bug?

edit `lib/reference/selectors.mjs`

<hr/>

caniuse is down  
ref https://developer.mozilla.org/en-US/docs/Web/CSS/:popover-open

**Comment by jeddy3** [comment]
SGTM.

[MDN reference for `popover-open`](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API)
